frontend/pages/admin/UserManagementPage/UserManagementPage.tsx

```diff
@@ -0,0 +1,643 @@
+          .catch(() => {
+            dispatch(
+              renderFlash(
+                "error",
+                `Could not edit ${userEditing?.name}. Please try again.`
+              )
+            );
+          })
+          .finally(() => {
+            toggleEditUserModal();
+          })
+      );
+    }
+
+    if (currentUser?.id === userEditing.id) {
+      return (
+        userData &&
+        usersAPI
+          .update(userData.id, formData)
+          .then(() => {
+            dispatch(
+              renderFlash("success", `Successfully edited ${userEditing?.name}`)
+            );
+          })
+          .then(() => refetchUsers())
+          .catch(() => {
+            dispatch(
+              renderFlash(
+                "error",
+                `Could not edit ${userEditing?.name}. Please try again.`
+              )
+            );
+          })
+          .finally(() => toggleEditUserModal())
+      );
+    }
+
+    let userUpdatedFlashMessage = `Successfully edited ${formData.name}`;
+
+    if (userData?.email !== formData.email) {
+      userUpdatedFlashMessage += `: A confirmation email was sent from ${config?.sender_address} to ${formData.email}`;
+    }
+
+    return (
+      userData &&
+      usersAPI
+        .update(userData.id, formData)
+        .then(() => {
+          dispatch(renderFlash("success", userUpdatedFlashMessage));
+        })
+        .then(() => refetchUsers())
+        .catch(() => {
+          dispatch(
+            renderFlash(
+              "error",
+              `Could not edit ${userEditing?.name}. Please try again.`
+            )
+          );
+        })
+        .finally(() => {
+          toggleEditUserModal();
+        })
+    );
+  };
+
+  const onDeleteUser = () => {
+    if (userEditing.type === "invite") {
+      invitesAPI
+        .destroy(userEditing.id)
+        .then(() => {
+          dispatch(
+            renderFlash("success", `Successfully deleted ${userEditing?.name}.`)
+          );
+        })
+        .catch(() => {
+          dispatch(
+            renderFlash(
+              "error",
+              `Could not delete ${userEditing?.name}. Please try again.`
+            )
+          );
+        })
+        .finally(() => {
+          toggleDeleteUserModal();
+          refetchInvites();
+        });
+    } else {
+      usersAPI
+        .destroy(userEditing.id)
+        .then(() => {
+          dispatch(
+            renderFlash("success", `Successfully deleted ${userEditing?.name}.`)
+          );
+        })
+        .catch(() => {
+          dispatch(
+            renderFlash(
+              "error",
+              `Could not delete ${userEditing?.name}. Please try again.`
+            )
+          );
+        })
+        .finally(() => {
+          toggleDeleteUserModal();
+          refetchUsers();
+        });
+    }
+  };
+
+  const onResetSessions = () => {
+    const isResettingCurrentUser = currentUser?.id === userEditing.id;
+
+    usersAPI
```

Comment:
useQuery?

Reply:
Frontend pattern is useQuery for GET requests only